test/unit/group-controller.test.js

```diff
@@ -0,0 +1,303 @@
+import 'babel-polyfill';
+import { v4 as uuidv4 } from 'uuid';
+import GroupService from '../../src/service/group-service';
+import {
+  addGroup, addUsersToGroup, getGroups, getGroupById, updateGroup, deleteGroup,
+} from '../../src/controllers/group-controller';
+
+jest.mock('../../src/service/group-service', () => ({
+  getAll: jest.fn(),
+  getById: jest.fn(),
+  addUsersToGroup: jest.fn(),
+  add: jest.fn(),
+  update: jest.fn(),
+  delete: jest.fn(),
+}));
+
+
+describe('group controller', () => {
+  let response;
+
+  beforeEach(() => {
+    response = {
+      status: jest.fn(() => response),
+      json: jest.fn(),
+      send: jest.fn(),
+    };
+  });
+
+  afterEach(() => {
+    jest.clearAllMocks();
+  });
+
+  test('gets all groups', async () => {
+    const request = {};
+
+    response.status.mockReturnValueOnce(response);
+    const next = jest.fn();
+
+    const allGroups = [{ name: 'group' }];
+    GroupService.getAll.mockResolvedValueOnce(allGroups);
+
+    await getGroups(request, response, next);
+
+    expect(response.status).toHaveBeenCalledWith(200);
+    expect(response.send).toHaveBeenCalledTimes(1);
+    expect(response.send.mock.calls.length).toBe(1);
+    expect(response.send).toHaveBeenCalledWith({ groups: allGroups });
+  });
+
+  test('gets all groups if groups are not exist', async () => {
```

Comment:
If you test the same method but with different scenarios it is better to wrap them with a separate `describe` block (inner describe block).
Like this and previous one. So it will remove duplicated part from test descriptions, as common part will be in `describe` description.

```
describe('group controller', () => { // outer describe block

   describe('get all groups method', ()=>{ // inner describe block

      test('should return all groups', ()=>{ ... }) // specific test
      test('should return empty array if there are no groups', ()=>{ ... }) // specific test

   })

});

```

You can also use separate beforeEach, beforeAll methods in inner describe blocks to initialize specific data.
Here you can find the order of execution: https://jestjs.io/docs/en/setup-teardown#order-of-execution-of-describe-and-test-blocks